tests/tests/test_middleware.py

```diff
@@ -380,3 +378,70 @@ def test_no_locking_on_store(self, client):
         assert request.idempotency_key_exempt is False
         assert request.idempotency_key_manual is False
         assert request.idempotency_key_encoded_key == 'f7a64a46c05113ce5828b8df7230c27e19e5934419c07b2feed9a52ba7bdbd5a'
+
+    @override_settings(
+        IDEMPOTENCY_KEY={
+            'STORAGE': {
+                'CLASS': 'idempotency_key.storage.CacheKeyStorage',
+                'CACHE_NAME': 'SevenDayCache',  # This should be overridden by the decorator
+            },
+        },
+    )
+    def test_middleware_cache_storage_using_custom_cache_name_on_decorator(self, client):
+        """
+        Tests @idempotency_key(cache_name='FiveMinuteCache') decorator
+        """
+        caches['default'].clear()
```

Comment:
be careful with this, this could affect all tests using `caches` setting. You could use fixtures to change the value set the original one.

Reply:
Whenever a test is run I want all related caches cleared. I had an issue with tests failing because a key still existed from the previous test.